### PR TITLE
Adding Gamepad to support controllers in both mobile and desktop modes

### PR DESCRIPTION
**Pull request by xesf**
Support for gamepads. Not merging yet because I want to test this better on mobile, but I need a PR since I cannot test the controller while the phone is connect to the laptop. You can still review it though.

**Preview here:** https://pr-427.lba2remake.net

### CODE REVIEW

File: src/controls/index.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { VRControls } from './vr';
 import { SceneManager } from '../game/SceneManager';
 import Renderer from '../renderer';
 import Game from '../game/Game';
+import GamepadManager from './Gamepad';
```
Comment (agrande) [inline src/controls/index.ts:9]: Why use a different name for the file and class?
Why not calling both GamepadControls, to follow the same nomenclature as the rest?

File: src/controls/keyboard.ts [hunk 2]
```diff
@@ -224,8 +225,7 @@ function focusOutHandler(game) {
     game.resetControlsState();
 }
 
-function resetCameraOrientation(game, sceneManager) {
-    const scene = sceneManager.getScene();
+export function resetCameraOrientation(game, scene) {
```
Comment (agrande) [inline src/controls/keyboard.ts:228]: Can you add some type annotations on game and scene?
We'd better do this from now on, on known types.

**Review by agrande — COMMENTED** [review]
You may want to try enabling controlsState.relativeToCam, it will probably feel more natural with a gamepad. I use this for VR.
Although it would require switching it on/off dynamically wether we used the keyboard or gamepad.

File: src/controls/index.ts (hunk 1 above)
Comment (xesf) [reply src/controls/index.ts:9]: Yeah, need to change it. Initially was supposed to be just Gamepad but that collides with the native API.
I will rename it later.

File: src/controls/keyboard.ts (hunk 2 above)
Comment (xesf) [reply src/controls/keyboard.ts:254]: Sounds good, been doing this for missed that one.

**Comment by xesf** [comment]
> You may want to try enabling controlsState.relativeToCam, it will probably feel more natural with a gamepad. I use this for VR.
> Although it would require switching it on/off dynamically wether we used the keyboard or gamepad.

Ok, will have a look to that one.
The gamepad fills a bit off and for some reasons when trying to test on mobile is not working, but locally works fine.

**Comment by xesf** [comment]
@slwilliams @agrande can you have another look to the PR?
I've added loads of changes since the first version but everything should be there now except the Inventory navigation that can be added when Scott finishes the inventory piece.

There is one annoying bug though. I need to look further.
The left stick X axis, when the value is very very low, close to zero, but still positive, it acts as Negative.

**Comment by agrande** [comment]
Lgtm, but I want to check it after I finish implementing what I'm doing on VR selector screen.

File: src/controls/GamepadControls.ts [hunk 3]
```diff
@@ -0,0 +1,260 @@
+import Game from '../game/Game';
+import { SceneManager } from '../game/SceneManager';
+import { Params } from '../params';
+import { BehaviourMode } from '../game/loop/hero';
+import { resetCameraOrientation } from './keyboard';
+import { ControlsState, ControlActiveType } from '../game/ControlsState';
+// import { resetCameraOrientation } from './keyboard';
```
Comment (agrande) [inline src/controls/GamepadControls.ts:7]: If you're not gonna use this in the end, maybe remove that line?

**Comment by xesf** [comment]
Merged with master for the VR selection fixes

**Review by agrande — APPROVED** [review]
lgtm

**Pull request merged**